authentication/account/repository/identity_blackbox_test.go

```diff
@@ -43,6 +45,55 @@ func (s *IdentityRepositoryTestSuite) TestDelete() {
 		}
 	})
 
+	s.T().Run("ok hard delete by identity ID", func(t *testing.T) {
+		// given
+		g := s.NewTestGraph(t)
+		identity1 := g.CreateIdentity()
+		// create a second identity
+		g.CreateIdentity()
+
+		unscoped := func(s *gorm.DB) *gorm.DB {
+			return s.Unscoped()
+		}
+		// when hard delete user
+		err := s.Application.Identities().Delete(s.Ctx, identity1.ID(), unscoped)
+		// then
+		require.Nil(t, err)
+
+		identity, err := s.Application.Identities().Load(s.Ctx, identity1.ID(), unscoped)
+		require.EqualError(t, err, fmt.Sprintf("identity with id '%s' not found", identity1.ID()))
+		require.Nil(t, identity)
+
+		identities, err := s.Application.Identities().List(s.Ctx)
```

Comment:
We don't really need to use List() here, do we? We need to pass scopes for `Delete()` to be able to hard delete identities. We need to pass scopes to `Load()` to test the record has been deleted. But we don't have to add it to List(), do we?
I would suggest not to modify List() at all. In your tests just check that the record is not found for both `Load(s.Ctx, identity1.ID(), unscoped)` and `Load(s.Ctx, identity1.ID())`.
And the same for User.

Reply:
Addressed in 72b2576